File: Concentus.Silk/Structs/silk_shape_state.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Concentus.Silk.Structs
{
    /// <summary>
    /// Noise shaping analysis state
    /// </summary>
    public class silk_shape_state
    {
        public sbyte LastGainIndex = 0;
        public int HarmBoost_smth_Q16 = 0;
        public int HarmShapeGain_smth_Q16 = 0;
        public int Tilt_smth_Q16 = 0;

        public void Reset()
        {
            LastGainIndex = 0;
            HarmBoost_smth_Q16 = 0;
            HarmShapeGain_smth_Q16 = 0;
            Tilt_smth_Q16 = 0;
        }
    }
}

File: Go/opus/silk_ShapeState.go
package opus

type SilkShapeState struct {
	LastGainIndex          int8
	HarmBoost_smth_Q16     int
	HarmShapeGain_smth_Q16 int
	Tilt_smth_Q16          int
}

func (s *SilkShapeState) Reset() {
	s.LastGainIndex = 0
	s.HarmBoost_smth_Q16 = 0
	s.HarmShapeGain_smth_Q16 = 0
	s.Tilt_smth_Q16 = 0
}
